build.gradle

```diff
@@ -119,6 +118,25 @@ allprojects {
     }
 }
 
+/**
+ * An argument provider which avoids task input inconsistency when builds are run
+ * from different locations. By handling this, we can take advantage of build caches
+ * when run from different locations. (i.e. remote build caches)
+ */
+class TestBuildDirArgumentsProvider implements CommandLineArgumentProvider {
```

Comment:
Can we move this to `/buildSrc` so we can reuse it in other places in our build?